timemachine/fe/mcgregor.py

```diff
@@ -212,7 +224,12 @@ def mcs(
         # f"==FAILED==[NODES VISITED {mcs_result.nodes_visited} | time taken: {time.time()-start_time} | time out? {mcs_result.timed_out}]====="
         # )
 
-    assert len(mcs_result.all_maps) > 0
+        # If timed out and no maps found, raise exception.
+        if mcs_result.timed_out:
```

Comment:
It seemed like if you had found a map(s) and timed out, was worth returning it rather then raising the error. Perhaps worth having a warning if there is a map and it timed out? But it seems possible to find a map and still end up in a pretty large search space that you might not want to completely explore for performance reasons.

Reply:
Are you seeing this in practice? It was an assert b/c if the result timed out we're not sure how to interpret the mapping, i.e. it may or may not be optimal. There were no timeouts in all of the test cases that I tried.

Reply:
I had not. I would agree it may not be optimal, it wasn't clear on if we want the true optimal every time. Just seemed like you might want to bound the search space and get back a mapping even if it isn't perfect.